with open() as f:
    ...
    return
try:
    f = open()
    ...
    return
except:
    print "error"
finally:
    f.close()

cout << "a"
..
cout << flush